std-bits/table/src/main/java/org/enso/table/data/column/builder/InferredBuilder.java

```diff
@@ -61,6 +63,11 @@ public void appendNoGrow(Object o) {
 
   @Override
   public void append(Object o) {
+    if (o instanceof Value v) {
+      // ToDo: This a workaround for an issue with polyglot layer. #5590 is related.
+      o = Polyglot_Utils.convertPolyglotValue(v);
+    }
```

Comment:
```suggestion
    // ToDo: This a workaround for an issue with polyglot layer. #5590 is related.
    o = Polyglot_Utils.convertPolyglotValue(v);
```

The `if` is not necessary I think? Because `Polyglot_Utils.convertPolyglotValue` has an overload for `Object` that does _exactly_ that.